baybe/parameters/enum.py

```diff
@@ -17,17 +19,39 @@ class CategoricalEncoding(ParameterEncoding):
     """Integer encoding."""
 
 
-class SubstanceEncoding(ParameterEncoding):
-    """Available encodings for substance parameters."""
-
-    MORDRED = "MORDRED"
-    """Encoding based on Mordred chemical descriptors."""
-
-    RDKIT = "RDKIT"
-    """Encoding based on RDKit chemical descriptors."""
-
-    MORGAN_FP = "MORGAN_FP"
-    """Encoding based on Morgan molecule fingerprints."""
+# TODO Ideally, this should be turned into a class that can:
+#  - return default when CHEM not installed
+#  - check if enum is fingerprint
+PARAM_SUFFIX_FINGERPRINT = "Fingerprint"
+
+if CHEM_INSTALLED:
+    import inspect
+
+    from baybe._optional.chem import BaseFingerprintTransformer, skfp_fingerprints
+
+    AVAILABLE_SKFP_FP = dict(
+        inspect.getmembers(
+            skfp_fingerprints,
+            lambda x: inspect.isclass(x) and issubclass(x, BaseFingerprintTransformer),
+        )
+    )
+    AVAILABLE_SKFP_FP["Default"] = AVAILABLE_SKFP_FP["MordredFingerprint"]
+else:
+    AVAILABLE_SKFP_FP = {"Default": None}
+
+AVAILABLE_SKFP_FP = {
+    (
+        name
+        if name.endswith(PARAM_SUFFIX_FINGERPRINT)
+        else name + PARAM_SUFFIX_FINGERPRINT
+    ): fp
+    for name, fp in AVAILABLE_SKFP_FP.items()
+}
+
+SubstanceEncoding = ParameterEncoding(
+    value="SubstanceEncoding", names={k: k for k in AVAILABLE_SKFP_FP.keys()}
```

Comment:
prob wont work forget about it I guess we should try simple first: hardcoded enum + lookup dict

Reply:
yes, I had str:class before but that messed up a lot of code elsewhere - if I  remember correctly since other parts of code base expect this to be string and checks for whether this is FP class require chem import etc